adapter/service/mediaplayer/service.go

```diff
@@ -0,0 +1,315 @@
+package mediaplayer
+
+import (
+	"fmt"
+	"slices"
+	"sync"
+
+	"github.com/futurehomeno/fimpgo"
+	"github.com/futurehomeno/fimpgo/fimptype"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/futurehomeno/cliffhanger/adapter"
+	"github.com/futurehomeno/cliffhanger/adapter/cache"
+)
+
+const (
+	// PropertySupportedPlayback is a property representing supported playback actions.
+	PropertySupportedPlayback = "sup_playback"
+	// PropertySupportedModes is a property representing supported playback modes.
+	PropertySupportedModes = "sup_modes"
+	// PropertySupportedMetadata is a property representing supported metadata.
+	PropertySupportedMetadata = "sup_metadata"
+)
+
+// DefaultReportingStrategy is the default reporting strategy used by the service for periodic reports.
+var DefaultReportingStrategy = cache.ReportOnChangeOnly()
+
+// Controller is an interface representing an actual device.
+type Controller interface {
+	// SetPlayback sets the playback state.
+	SetPlayback(action string) error
```

Comment:
As actions, modes and metadata fields are strictly defined, I guess we can stick to separate types & defined constants for them.